### PR TITLE
[20725] Add new watcher handling

### PR DESCRIPTION
**Pull request by floriank**
This PR will add improved handling for the watchers tab in the split screen view.

It should cover the needs of https://community.openproject.org/work_packages/20725.
## Things to discuss/todo:
- **TESTS**
  - [x] basic tests for services and controller
- [x] Styling is not done at all (cc @HDinger)
- [x] String are not translated at the moment

As always, feedback welcome

### CODE REVIEW

**Comment by TeatroIO** [comment]
I've prepared a stage to preview changes. <a href='http://floriank-feature-20725-le-awesome-watchers-implementation.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com' target='_blank'>Open stage</a> or <a href='http://floriank-feature-20725-le-awesome-watchers-implementation.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com/teatro.log' target='_blank'>view logs</a>.

File: frontend/app/templates/work_packages/watchers/watcher.html [hunk 1]
```diff
@@ -0,0 +1,7 @@
+<li data-ng-class="{'loading':watcher.loading,'deleting':deleting}">
+  <a ng-href="{{watcher._links.self.href}}" title="{{watcher.name}}">
```
Comment (floriank) [inline frontend/app/templates/work_packages/watchers/watcher.html]: **NOTE** to myself: this is the API response and **not** the users profile page

**Comment by floriank** [comment]
@ulferts @basiszwo I could add tests for integration and directives, however that will take time - should I?

**Comment by floriank** [comment]
@HDinger I'd like to delegate styling to you - if you have questions, we can talk.

**Comment by floriank** [comment]
@opf/developers In general, I'd like feedback on this, I chose a more pragmatical approach in implementation as time is limited, so I could use all of your input.

**Comment by floriank** [comment]
Notes from @basiszwo (via email, yesterday evening):
- watcher selection should be a `ui.select`, like the assignee select
  - on click the list of available users should be shown (default behaviour)
- the mini menu (like with the other in place edit fields should be shown to confirm the selection, but without the "send email" button
- the user is added above in the list (**NOTE to self:** _Appended_ to the list above, not _prepended_)

**Comment by ulferts** [comment]
@HDinger I have the impression that the height of the watchers is less than the height of the inplace edit fields. Can you make them have the same height?

Watchers
image
vs Inplace edit fields:
image

File: app/assets/stylesheets/content/work_package_details/_watchers_tab.sass [hunk 2]
```diff
@@ -38,6 +36,9 @@
   input[type='text']
     width: 100%
```
Comment (ulferts) [inline app/assets/stylesheets/content/work_package_details/_watchers_tab.sass]: Do we still need all those styles @HDinger. Especially the select2/ui-select once I assume to be unnecessary by now.

**Comment by ulferts** [comment]
Just a note to self or whoever will take over the branch. It would probably be good to double check (ask the user) before deleting an attachment. I don't think that it is necessary for watchers as the user is not actually deleted, but an attachment is gone once the user clicks the icon.

File: frontend/webpack.config.js [hunk 3]
```diff
@@ -124,7 +124,8 @@ module.exports = {
       'hyperagent': 'hyperagent/dist/hyperagent',
       'openproject-ui_components':
         'openproject-ui_components/app/assets/javascripts/angular/ui-components-app',
-      'ngFileUpload': 'ng-file-upload/ng-file-upload'
+      'ngFileUpload': 'ng-file-upload/ng-file-upload',
+      'angucomplete-alt': 'angucomplete-alt/angucomplete-alt'
```
Comment (ulferts) [inline frontend/webpack.config.js:128]: This is a relic introduced in a9538ca3 when `angucomplete` was still used in lookup.html.

**Comment by HDinger** [comment]
During the styling some functionality got lost in the attachment field. At the moment you cannot load the attachment by directly clicking on it.

File: frontend/webpack.config.js (hunk 3 above)
Comment (floriank) [inline frontend/webpack.config.js:128]: This should be easily removable as I already removed the dependencies in `openproject-core-app.js`

**Comment by ulferts** [comment]
Superseeded by #3434

**Pull request closed without merging**